.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,70 @@
+name: Unit tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  checks:
+    name: Run clippy and unit tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all
```

Comment:
I think we need to pass `--check` as well (see https://github.com/rust-lang/rustfmt#checking-style-on-a-ci-server)

Reply:
done